src/containers/MyNdla/components/MenuModalContent.tsx

```diff
@@ -0,0 +1,198 @@
+/**
+ * Copyright (c) 2023-present, NDLA.
+ *
+ * This source code is licensed under the GPLv3 license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import { css } from '@emotion/react';
+import styled from '@emotion/styled';
+import { IconButtonV2 } from '@ndla/button';
+import { spacing, fonts, colors } from '@ndla/core';
+import { FourlineHamburger, List } from '@ndla/icons/action';
+import {
+  ModalBody,
+  ModalHeader,
+  ModalContent,
+  ModalCloseButton,
+} from '@ndla/modal';
+import { ReactNode, useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useLocation, useOutletContext } from 'react-router-dom';
+import { ViewType } from '../Folders/FoldersPage';
+import { OutletContext, menuActions } from '../MyNdlaLayout';
+import NavigationLink from './NavigationLink';
+
+const MenuItem = styled.li`
+  list-style: none;
+  flex-basis: 24%;
+  margin: unset;
+`;
+
+const Title = styled.div`
+  text-transform: uppercase;
+  padding: ${spacing.normal} ${spacing.small} ${spacing.small} ${spacing.small};
+  ${fonts.sizes('18px', '24px')};
+  font-weight: ${fonts.weight.bold};
+  color: ${colors.brand.grey};
+
+  &[data-border-top='true'] {
+    border-top: 1px solid ${colors.brand.lightest};
+  }
+
+  &[data-no-padding-top='true'] {
+    padding-top: unset;
+  }
+
+  &[data-blue-background='true'] {
+    background: ${colors.background.lightBlue};
+  }
+`;
+
+const MenuItems = styled.ul`
+  display: flex;
+  flex-direction: row;
+  flex-wrap: wrap;
+
+  justify-content: space-between;
+  padding: unset;
+  margin: unset;
+  padding-bottom: ${spacing.small};
+  border-bottom: 1px solid ${colors.brand.lighter};
+
+  background: ${colors.background.lightBlue};
+`;
+
+const ToolItem = css`
+  border-top: 0.5px solid ${colors.brand.greyLighter};
+  padding: ${spacing.small} ${spacing.normal};
+  &:last-child {
+    border-bottom: 0.5px solid ${colors.brand.greyLighter};
+  }
+`;
+
+const ToolMenu = styled.div`
+  display: flex;
+  flex-direction: column;
+
+  padding: unset;
+  margin: unset;
+
+  button,
+  a {
+    ${ToolItem};
+  }
```

Comment:
Dette blir gjort for å unngå at `FoldersPage.tsx` blir 500 linjer lang. Det er så mange knapper nå som skal rendres ved ulike scenarioer som skaper mange ekstra linjer kode. 

 Ikke 100% fan av dette da det kan bli rotete å finne fram til ved debug.. men fant ingen bedre løsning som gjorde det mulig å flytte all knapp logikk ut av FoldersPage.  Ved å flytte ut knappe logikken fra `FoldersPage` ble filen mye enklere å forholde seg til, føler uansett at med antallet knapper vi har nå så burde de blitt separert til egen fil. 

En negativ med  endringen  er at  det da er umulig å iterere over knappene for rendre dem inn i en ul  wrappet av li element, noe som de burde være da knappene er en liste av valg du kan gjøre på en valgt mappe.

Et positiv er at de gir en større trykk overflate på mobil, som jeg er mer positiv til enn liste problemet.